addons/binding/org.openhab.binding.hpprinter/src/main/java/org/openhab/binding/hpprinter/internal/handler/WebPutter.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hpprinter.internal.handler;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Base64;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.hpprinter.internal.config.HPPrinterConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Performs web puts to EWS.
+ *
+ * @author Stewart Cossey - Initial contribution
+ */
+@NonNullByDefault
+public class WebPutter {
+
+    private final Logger logger = LoggerFactory.getLogger(HPPrinterHandler.class);
+
+    private final HPPrinterConfiguration config;
+    private final String endpoint;
+
+    WebPutter(String endpoint, HPPrinterConfiguration config) {
+        this.endpoint = endpoint;
+        this.config = config;
+    }
+
+    private String authHeader() {
+        logger.trace("PUT Request using auth");
+        String combined = config.userName + ":" + config.ewsPassword;
+        String basicAuth = "Basic " + new String(Base64.getEncoder().encode(combined.getBytes()));
+        return basicAuth;
+    }
+
+    public boolean put(String outputText) {
+        HttpsURLConnection putSecureRequestConn = null;
+        HttpURLConnection putRequestConn = null;
+        DataOutputStream dataOutputStream = null;
+
+        try {
+            URL url = new URL(config.useSSL ? "https" : "http", config.ipAddress, endpoint);
+
+            if (config.useSSL) {
+                putSecureRequestConn = (HttpsURLConnection) url.openConnection();
```

Comment:
Please take a look at removing some duplication between this block and the block starting at line 78